.circleci/config.yml

```diff
@@ -1,121 +1,218 @@
       - run:
-          name: 🚧 install dependencies from latest master commit
+          name: 🚧 build misc
           command: |
-              git clone --depth 1 https://github.com/plotly/dash-core-components.git
-              git clone --depth 1 https://github.com/plotly/dash-html-components.git
+              . venv/bin/activate && pip install --upgrade -e . --quiet && mkdir packages
               git clone --depth 1 https://github.com/plotly/dash-table.git
-              git clone --depth 1 https://github.com/plotly/dash-renderer-test-components
-              . venv/bin/activate
-              cd dash-core-components && npm install --ignore-scripts && npm run build && pip install -e . && cd ..
-              cd dash-html-components && npm install --ignore-scripts && npm run build && pip install -e . && cd ..
-              cd dash-table && npm install --ignore-scripts && npm run build && pip install -e . && cd ..
-              cd dash-renderer-test-components && npm install --ignore-scripts && npm run build:all && pip install -e . && cd ..
+              cd dash-table && npm install --ignore-scripts && npm run build && python setup.py sdist && mv dist/* ../packages/ && cd ..
+              git clone --depth 1 https://github.com/plotly/dash-html-components.git
+              cd dash-html-components && npm install --ignore-scripts && npm run build && python setup.py sdist && mv dist/* ../packages/ && cd ..
+      - persist_to_workspace:
+          root: ~/dash
+          paths:
+            - packages
 
+  "build-misc-36":
+      <<: *build-misc
+      docker:
+        - image: circleci/python:3.6-stretch-node-browsers
+          environment:
+            PYVERSION: python36
+
+  "build-misc-27":
+      <<: *build-misc
+      docker:
+        - image: circleci/python:2.7-stretch-node-browsers
+          environment:
+            PYVERSION: python27
+
+  "test-37": &test
+    working_directory: ~/dash
+    docker:
+      - image: circleci/python:3.7-stretch-node-browsers
+        environment:
+          PERCY_PARALLEL_TOTAL: '-1'
+          PYVERSION: python37
+    parallelism: 3
+    steps:
+      - checkout
+      - run: echo $PYVERSION > ver.txt
+      - restore_cache:
+          key: dep-{{ checksum "ver.txt" }}-{{ checksum "requires-ci.txt" }}-{{ checksum "requires-install.txt" }}-{{ checksum "requires-testing.txt" }}
+      - attach_workspace:
+          at: ~/dash
       - run:
-          name: ⚙️ run integration test
+          name: ⚙️ Run Integration Tests
           command: |
             . venv/bin/activate
-            pytest --junitxml=test-reports/junit_intg.xml tests/integration/
+            cd packages && ls -la && pip install * && cd .. && pip list | grep dash
+            TESTFILES=$(circleci tests glob "tests/integration/**/test_*.py" | circleci tests split --split-by=timings)
```

Comment:
circleci way to split and select tests based on previous execution duration infos